dev/com.ibm.ws.webcontainer.servlet.3.1_fat.2/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/CustomizedExceptionText.java

```diff
@@ -0,0 +1,146 @@
+    }
+
+    @Test
+    public void test500CustomMessage() throws Throwable {
+        LOGGER.info("Starting test500CustomMessage");
+        updateCustomizedExceptionText("Custom 500 internal server error message");
+
+        URL requestURL = HttpUtils.createURL(server, "/" + CUSTOM_ERROR_APP_NAME + "/error");
+        String result = checkRequest(requestURL.toString(), 500, "Custom 500 internal server error message", "SRVE0232E");
+
+        assertTrue("Custom 500 text not found in response:\n" + result, result.contains("PASS"));
+    }
+
+    private String checkRequest(String URL, int responseCode, String expectedResText, String notExpectedResText)throws Throwable {
+        WebConversation wc = new WebConversation();
+        wc.setExceptionsThrownOnErrorStatus(false);
+        WebResponse resp = wc.getResponse(URL);
+
+        int code = resp.getResponseCode();
+        String body = resp.getText();
+
+        StringBuilder result = new StringBuilder();
+        result.append("Checking Response Code: Expected [" + responseCode + "], actual [" + code + "]. ");
+        result.append(code == responseCode ? "PASS\n" : "FAIL\n");
+
+        if (!expectedResText.equals("SKIP")) {
+            result.append("Checking Expected Response text: ");
+            result.append(body.contains(expectedResText) ? "PASS\n" : "FAIL\n");
+        }
+
+        if (!notExpectedResText.equals("SKIP")) {
+            result.append("Checking NOT Expected Response text: ");
+            result.append(body.contains(notExpectedResText) ? "FAIL\n" : "PASS\n");
+        }
+
+        LOGGER.log(Level.INFO, "checkRequest Body:\n" + body);
+        return result.toString();
+    }
+
+    /**
+     * Method to update <webContainer displayCustomizedExceptionText="..." dynamically
+     */
+    private void updateCustomizedExceptionText(String customMessage) throws Exception {
+        ServerConfiguration config = server.getServerConfiguration();
+        WebContainerElement webContainer = config.getWebContainer();
+
+        boolean serverXMLChanged = false;
+        String currentValue = webContainer.getDisplayCustomizedExceptionText();
+
+        if (currentValue == null || !currentValue.equals(customMessage)) {
+            LOGGER.info("Updating displayCustomizedExceptionText from [" + currentValue + "] to [" + customMessage + "]");
+            webContainer.setDisplayCustomizedExceptionText(customMessage);
+            serverXMLChanged = true;
+        } else {
+            LOGGER.info("No update required value already set to [" + currentValue + "]");
+        }
+        server.setMarkToEndOfLog();
+        server.updateServerConfiguration(config);
+
+        // Wait for a config update if we actually changed something
+        if (serverXMLChanged) {
+            try {
+                server.waitForConfigUpdateInLogUsingMark(Collections.emptySet(),false,"CWWKG0017I|CWWKF0008I|CWWKT0016I:.*CustomErrorTestApp.*");
```

Comment:
As far as I know `waitForConfigUpdateInLogsUsingMark()` will already check for `CWWKG0017I` and `CWWKF0008I` doesn't it?